src/env.rs

```diff
@@ -122,6 +163,11 @@ impl Source for Environment {
             };
 
             m.insert(key, Value::new(Some(&uri), value));
+        };
+
+        match &self.source {
+            Some(source) => source.clone().into_iter().for_each(collector),
```

Comment:
do we actually need to `clone()` here? AFAICS, the collector could use references for the key-value-pair, couldn't it? Or would that conflict with the `env::vars()` iterator, because that one returns `(String, String)`?

Reply:
So `env::vars()` provides `(String,String)`, and some paths in the (now) closure consume the string value.

The `self.source` does not get moved, as it is part of `self`. I tried with a closure of `(AsRef<str>, Into<String>)`, however, this doesn't work because both are not `Sized`. That would require introducing a new function with type arguments. As this is code intended for testing, I think having the `clone` is ok. Besides, the `env::vars()` also clones, but hides this.